src/location/plugins/declarative_style_parameter.hpp

```diff
@@ -10,7 +10,12 @@
 #include <QtCore/QMetaObject>
 #include <QtCore/QMetaProperty>
 
-#define MLN_DECLARATIVE_PARSER(Class)                                                                           \
+#define MLN_DECLARATIVE_PARSER(Class) \
```

Comment:
I don't believe this macro is worth it.

Reply:
In most cases there is no custom init method needed. Is there a possibility for an "ifdef" so I only call the second parameter if available?

Reply:
Can't you use inheritance for componentComplete?

Reply:
Let me try again, but I remember that I was not possible (i.e. `QQmlParserStatus` needs to be direct base class).

Reply:
OK, I found a cleaner solution -> wrap the property access and convert if needed.